backtrace-library/src/main/java/backtraceio/library/metrics/BacktraceMetrics.java

```diff
@@ -0,0 +1,309 @@
+
+    /**
+     * Default time interval in min
+     */
+    public final static int defaultTimeIntervalInMin = 30;
+
+    /**
+     * Default time interval in milliseconds
+     */
+    public final static long defaultTimeIntervalMillis = defaultTimeIntervalInMin * 60 * 1000;
+
+    /**
+     * Default unique event name that will be generated on app startup
+     */
+    public final String defaultUniqueEventName = "guid";
+
+    /**
+     * Name of the unique event that will be generated on app startup
+     */
+    private String startupUniqueEventName = defaultUniqueEventName;
+
+    /**
+     * Maximum number of events in store. If number of events in store hit the limit
+     * BacktraceMetrics instance will send data to Backtrace.
+     */
+    private int maximumEvents = 350;
+
+    /**
+     * Name of the summed event that will be generated on app startup
+     */
+    private final String startupSummedEventName = "Application Launches";
+
+    /**
+     * Maximum number of attempts
+     */
+    public final static int maxNumberOfAttempts = 3;
+
+    /**
+     * Default time between retries in seconds
+     */
+    public final static int defaultTimeBetweenRetriesMillis = 10000;
+
+    /**
+     * Maximum time between requests in minutes
+     */
+    public final static int maxTimeBetweenRetriesMillis = 5 * 60 * 1000;
+
+    /**
+     * Default submission url
+     */
+    public final static String defaultBaseUrl = "https://events.backtrace.io/api";
+
+    /**
+     * User custom request method
+     */
+    private RequestHandler requestHandler = null;
+
+    /**
+     * Unique Events handler
+     */
+    public UniqueEventsHandler uniqueEventsHandler;
+
+    /**
+     * Summed Events handler
+     */
+    public SummedEventsHandler summedEventsHandler;
+
+    /**
+     * Custom attributes provided by the user to BacktraceBase
+     */
+    protected Map<String, Object> customAttributes;
+
+    /**
+     * The application context
+     */
+    protected Context context;
+
+    /**
+     * Backtrace metrics settings
+     */
+    protected BacktraceMetricsSettings settings;
+
+    /**
+     * Create new Backtrace metrics instance
+     * @param context                   Application context
+     * @param customAttributes          Backtrace client
+     * @param settings
+     */
+    public BacktraceMetrics(Context context, Map<String, Object> customAttributes, BacktraceMetricsSettings settings) {
+        this.context = context;
+        this.customAttributes = customAttributes;
+        this.settings = settings;
+    }
+
+    @Override
+    public void startMetricsEventHandlers(Api backtraceApi) {
+        uniqueEventsHandler = backtraceApi.enableUniqueEvents(context, customAttributes, settings);
+        summedEventsHandler = backtraceApi.enableSummedEvents(context, customAttributes, settings);
+    }
+
+    void setStartupUniqueEventName(String StartupUniqueEventName) {
```

Comment:
I think it should be: `startupUniqueEventName`

Reply:
Why shouldn't we use `set` prefix here like we do with other setters in this library?

Reply:
sorry @rqbacktrace  - I wanted to say parameter name should be `startupUniqueEventName` (lower case vs upper case)

Reply:
Ahhhhh sorry I see my typo now thank you @konraddysput !